gpm_api/tests/io/test_data_integrity.py

```diff
@@ -0,0 +1,84 @@
+from gpm_api.io import data_integrity as di
+from typing import List, Tuple
+import os
+import posixpath as pxp
+import ntpath as ntp
+
+
+def test_get_corrupted_filepaths(
+    local_filepaths_unix: List[str],
+    local_filepaths_windows: List[str],
+) -> None:
+    """Test get_corrupted_filepaths function"""
+
+    # Test that all paths are "corrupted" (in this case there is no data)
+    for abs_paths in [local_filepaths_unix, local_filepaths_windows]:
+        res = di.get_corrupted_filepaths(abs_paths)
+
+        assert len(abs_paths) == len(
+            res
+        ), "Corrupted paths array should be the same length as input paths"
+        assert abs_paths == res, "Corrupted paths array should be the same as input paths"
+
+    # TODO: Test an actual HDF5 for OSError and empty list (success)
```

Comment:
Just start the download of a file (like 2A-DPR) and interrupt the download after 2 seconds. You will have the test file to test this ;)